### PR TITLE
S4 us17 add testimony form

### PR DESCRIPTION
**Pull request by RaphaelBS-WCS**

### CODE REVIEW

**Review by Mcbess-san — DISMISSED** [review]

File: src/Controller/TestimonyController.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+     * Display home page
+     *
+     * @return string
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\RuntimeError
+     * @throws \Twig\Error\SyntaxError
+     */
+
+
+    public array $errors = [];
```
Comment (LucyMerlier) [inline src/Controller/TestimonyController.php:20]: you're not using this

File: src/Model/TestimoniesManager.php [hunk 2]
```diff
@@ -13,7 +13,7 @@ class TestimoniesManager extends AbstractManager
 
     public function insert(array $item): int
```
Comment (LucyMerlier) [inline src/Model/TestimoniesManager.php:14]: your parameter is called "item"?

File: src/View/Visitor/addTestimony/index.html.twig [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            {% for error in errors %}
+                <li class="errors">{{ error }}</li>
+            {% endfor %}
+            </ul>
+        {% endif %}
+        <div class="info">
+            <label for="name">Nom complet :</label>
+            <input class="name" type="text" name="name" placeholder="Nom complet" required>
+            <label for="mail">Adresse mail :</label>
+            <input class="mail" type="text" name="mail" placeholder="exemple@exemple.com" required>
```
Comment (LucyMerlier) [inline src/View/Visitor/addTestimony/index.html.twig:22]: not using type="mail"?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments + I hadn't seen it before but I don't think you need a full update method for testimonies in your testimony manager

File: src/Controller/TestimonyController.php (hunk 1 above)
Comment (RaphaelBS-WCS) [reply src/Controller/TestimonyController.php]: corrected

File: src/Model/TestimoniesManager.php (hunk 2 above)
Comment (RaphaelBS-WCS) [reply src/Model/TestimoniesManager.php]: corrected

File: src/View/Visitor/addTestimony/index.html.twig (hunk 3 above)
Comment (RaphaelBS-WCS) [reply src/View/Visitor/addTestimony/index.html.twig]: corrected

**Review by Erdnaxela45 — APPROVED** [review]
carré comme la Corée

File: src/View/Visitor/addTestimony/index.html.twig [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            {% for error in errors %}
+                <li class="errors">{{ error }}</li>
+            {% endfor %}
+            </ul>
+        {% endif %}
+        <div class="info">
+            <label for="name">Nom complet :</label>
+            <input class="name" type="text" name="name" placeholder="Nom complet" required>
+            <label for="mail">Adresse mail :</label>
+            <input class="mail" type="mail" name="mail" placeholder="exemple@exemple.com" required>
```
Comment (LucyMerlier) [inline src/View/Visitor/addTestimony/index.html.twig:22]: sorry my bad, typo in my last comment : the correct html input type is "email", not "mail"

**Review by LucyMerlier — CHANGES_REQUESTED** [review]

File: src/View/Visitor/addTestimony/index.html.twig (hunk 4 above)
Comment (RaphaelBS-WCS) [reply src/View/Visitor/addTestimony/index.html.twig]: no worries, iI would have checked this.

**Review by khadimb — APPROVED** [review]
good

**Review by LucyMerlier — APPROVED** [review]

**Pull request merged**